package.json

```diff
@@ -85,6 +86,7 @@
   },
   "dependencies": {
     "@coder/logger": "1.1.16",
+    "bcrypt": "^5.0.1",
```

Comment:
I don't think there's anything wrong with bcrypt, but was just curious, have we considered argon2? It's supposed to be a newer & better key derivation function: https://cryptobook.nakov.com/mac-and-key-derivation/argon2

Reply:
I went with the first recommendation from CodeQL (but I see argon2 is in there as well).

![image](https://user-images.githubusercontent.com/3806031/119406398-958c1000-bc97-11eb-8180-b50b4aca212c.png)

This is not my wheelhouse so easily convinced to switch to `argon2`. There's an npm package for it: https://www.npmjs.com/package/argon2

I'll let @oxy weigh in.

Reply:
It might also eliminate (or reduce work required) for https://github.com/cdr/code-server/issues/3432

Reply:
Sounds like we should go with `argon2` then with 2 votes for it. I'm all for making it better and it shouldn't be too hard to switch out `bcrypt` for `argon2`

Reply:
I spoke to soon 😅 `argon2` only supports async methods. The previous methods were sync. In order to use `argon2` we would have to rewrite a lot of code. I tried just now by making a bunch of functions `async` but it seems to have broken a lot of things 😅

So instead, I think we'll have to stay with `bcrypt` as part of this fix. But we can look into switching to `argon2` if we find `bcrypt` to not fulfill our needs. Sorry for the false hope!

Reply:
Ahh things are never as easy as they seem lmao. I thought that `const hash = await argon2.hash("password");` would turn it into a sync operation?

Reply:
I don't think the codebase is set up to use top-level `await` which means we have to use it in an `async` function, which means any function that calls it must also use `await` and also be converted to an `async` function unfortunately

Reply:
But I am take a second stab at refactoring to use `argon2` 🤞

Reply:
Ahhhhh. Good luck Joe! And thanks for trying, regardless of the outcome, it seems tough :)